### ISSUE TITLE
`make install` uses absolute path without $(DEST) to delete files outside of the staging directory

### ISSUE THREAD

**Issue by aryoda** [bit-team/backintime#1630]
I have introduced a clean-up code in the `Makefile` to remove installed renamed files which could cause havoc when a new BiT version is installed via `make install`.

@uffe has now reported that `make install` may cause an unwanted side effect ("messing outside the staging dir"):

https://aur.archlinux.org/pkgbase/backintime#comment-954580

```
#clean-up installed old files that were renamed or moved in later BiT versions
rm -f /etc/dbus-1/system.d/net.launchpad.backintime.serviceHelper.conf
rm: cannot remove '/etc/dbus-1/system.d/net.launchpad.backintime.serviceHelper.conf': Permission denied
make: *** [Makefile:14: install] Error 1
```

**Comment by aryoda** [comment]
The problematic part  in `qt/Makefile` (generated via `qt/configure`) is:

https://github.com/bit-team/backintime/blob/ffbf79cc5e5dd705121679153d4665d507060f1a/qt/configure#L150-L154

`Makefile`:
```
install:
	#clean-up installed old files that were renamed or moved in later BiT versions
	rm -f /etc/dbus-1/system.d/net.launchpad.backintime.serviceHelper.conf
	rm -f $(DEST)/share/backintime/plugins/qt4plugin.py
```

The problem is most probably only relevant if `make` is called a different installation destination (`make DESTDIR=... install`) where `DESTDIR` is part of `$(DEST)` then in the `Makefile`.

I suggest to change this line from

https://github.com/bit-team/backintime/blob/ffbf79cc5e5dd705121679153d4665d507060f1a/qt/configure#L152

into

`printf "\trm -f \$(DEST)/etc/dbus-1/system.d/net.launchpad.backintime.serviceHelper.conf\n" >> ${MAKEFILE}`

**Comment by aryoda** [comment]
I have prepared a fix (PR #1633) and tested it with `make DESTDIR="/home/user/temp/bit" install` before and after the fix and it works now (though using a `DESTDIR` makes it impossible to delete the installed file but this is the responsibility of the package maintainer then).

**Comment by ptilopteri** [comment]
* aryoda ***@***.***> [02-02-24 11:53]:
> I have prepared a fix (PR #1633) and tested it with `make
> DESTDIR="/home/user/temp/bit" install` before and after the fix and it
> works now (though using a `DESTDIR` makes it impossible to delete the
> installed file but this is the responsibility of the package maintainer
> then).
> 
> Message ID: ***@***.***>


#! /bin/bash

cd /data/build
rm -rf /data/build/backintime

git clone https://github.com/bit-team/backintime
cd /data/build/backintime
git submodule init ; git submodule update

cd /data/build/backintime/common
./configure && make
sudo make install

cd /data/build/backintime/qt
./configure && make
sudo make install

sudo rm -f rm -f /usr/share/backintime/plugins/qt4plugin.py

backintime --version



note: make test fails:

...................
----------------------------------------------------------------------
Ran 19 tests in 0.034s

OK
.....................
----------------------------------------------------------------------
Ran 21 tests in 3.757s

OK
QSettings::value: Empty key passed
QSettings::value: Empty key passed
..s
----------------------------------------------------------------------
Ran 3 tests in 5.600s

OK (skipped=1)
WARNING: Failed to read process stat from /proc/27562/stat: [2] No such file or directory
WARNING: Failed to read process stat from /proc/27563/stat: [2] No such file or directory
WARNING: Failed to read process stat from /proc/27564/stat: [2] No such file or directory
...............
----------------------------------------------------------------------
Ran 15 tests in 3.264s

OK
/usr/lib64/python3.11/subprocess.py:1127: ResourceWarning: subprocess 27608 is still running
WARNING: Failed to read process stat from /proc/28520/stat: [2] No such file or directory
......................................................
----------------------------------------------------------------------
Ran 54 tests in 0.014s

OK
...........
----------------------------------------------------------------------
Ran 11 tests in 0.396s

OK
E
======================================================================
ERROR: test_diagnostics (unittest.loader._FailedTest.test_diagnostics)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_diagnostics
Traceback (most recent call last):
  File "/usr/lib64/python3.11/unittest/loader.py", line 162, in loadTestsFromName
    module = __import__(module_name)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/build/backintime/common/test/test_diagnostics.py", line 4, in <module>
    import pyfakefs.fake_filesystem_unittest as pyfakefs_ut
ModuleNotFoundError: No module named 'pyfakefs'


----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)
make: *** [Makefile:568: unittest] Error 1





-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
http://en.opensuse.org    openSUSE Community Member    facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by aryoda** [comment]
> * aryoda ***@***.***> [02-02-24 11:53]:
> I have prepared a fix (PR #1633) and tested it with `make DESTDIR="/home/user/temp/bit" install` before and after the fix and it works now (though using a `DESTDIR` makes it impossible to delete the installed file but this is the responsibility of the package maintainer then). Message ID: ***@***.***>

```
ModuleNotFoundError: No module named 'pyfakefs'
```

@ptilopteri 

I had this unit test failure too.

It is a build/test dependency and you need to install `python311-pyfakefs` (or whichever python version you are using)

**Edit: ** Our [dependencies are documented here](https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#dependencies) but we do not provide distro packages so the package names are listed only for Debian and you need to find the corresponding OpenSuse package somehow...
